circuits/cpp/barretenberg/cpp/src/barretenberg/ecc/fields/field2_declarations.hpp

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include "barretenberg/numeric/uint256/uint256.hpp"
+
+// forward declare Engine
+namespace numeric::random {
+class Engine;
+}
+
+namespace barretenberg {
+template <class base_field, class Params> struct alignas(32) field2 {
+  public:
+    constexpr field2(const base_field& a = base_field::zero(), const base_field& b = base_field::zero())
+        : c0(a)
+        , c1(b)
+    {}
+
+    constexpr field2(const field2& other) noexcept
+        : c0(other.c0)
+        , c1(other.c1)
+    {}
+    constexpr field2(field2&& other) noexcept
+        : c0(other.c0)
+        , c1(other.c1)
+    {}
+
+    constexpr field2& operator=(const field2& other) noexcept
+    {
+        if (this == &other) {
+            return *this;
+        }
+        c0 = other.c0;
+        c1 = other.c1;
+        return *this;
+    }
+
+    constexpr field2& operator=(field2&& other) noexcept
+    {
+        if (this == &other) {
+            return *this;
+        }
+        c0 = other.c0;
+        c1 = other.c1;
+        return *this;
+    }
+
+    constexpr ~field2() noexcept = default;
+
+    base_field c0;
+    base_field c1;
+
+    static constexpr uint256_t modulus = base_field::modulus;
+
+    static constexpr field2 zero() { return field2{ base_field::zero(), base_field::zero() }; }
+    static constexpr field2 one() { return field2{ base_field::one(), base_field::zero() }; }
+    static constexpr field2 twist_coeff_b() { return field2{ Params::twist_coeff_b_0, Params::twist_coeff_b_1 }; }
+    static constexpr field2 twist_mul_by_q_x()
+    {
+        return field2{ Params::twist_mul_by_q_x_0, Params::twist_mul_by_q_x_1 };
+    }
+    static constexpr field2 twist_mul_by_q_y()
+    {
+        return field2{ Params::twist_mul_by_q_y_0, Params::twist_mul_by_q_y_1 };
+    }
+    static constexpr field2 cube_root_of_unity()
+    {
+        return field2{ Params::twist_cube_root_0, Params::twist_cube_root_1 };
+    }
+
+    constexpr field2 operator*(const field2& other) const noexcept;
+    constexpr field2 operator+(const field2& other) const noexcept;
+    constexpr field2 operator-(const field2& other) const noexcept;
+    constexpr field2 operator-() const noexcept;
+    constexpr field2 operator/(const field2& other) const noexcept;
+
+    constexpr field2 operator*=(const field2& other) noexcept;
+    constexpr field2 operator+=(const field2& other) noexcept;
+    constexpr field2 operator-=(const field2& other) noexcept;
+    constexpr field2 operator/=(const field2& other) noexcept;
+
+    constexpr field2 mul_by_fq(const base_field& a) const noexcept
+    {
+        field2 r{ a * c0, a * c1 };
+        return r;
+    }
+
+    // constexpr bool operator>(const field& other) const noexcept;
+    // constexpr bool operator<(const field& other) const noexcept;
```

Comment:
Don't think we should ever support this fwiw; there is no ordering for finite fields so this will likely cause bugs if ever implemented

Reply:
We have these operators in `field` because they are required for standard library methods like `std::sort`. I've removed the commented declarations in `field2` and added clarifying comments for the `field` class